src/EditorFeatures/Core.Wpf/InheritanceMargin/InheritanceMarginTag.cs

```diff
@@ -28,10 +26,13 @@ internal class InheritanceMarginTag : IGlyphTag
         /// </summary>
         public readonly ImmutableArray<InheritanceMarginItem> MembersOnLine;
 
-        public InheritanceMarginTag(ImmutableArray<InheritanceMarginItem> membersOnLine)
+        public readonly Document Document;
```

Comment:
why do you need to hold onto the document?

Reply:
It is used when call this http://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/GoToDefinition/GoToDefinitionHelpers.cs,117

I look into it and seems like it only needs Workspace. Would referencing documents cause any problem (memory? perf?)

Reply:
quite possibly yes.  general concern is some UI element holding onto an old snapshot for too long, keeping large swaths of objects rooted and not being GC'ed.  If you can avoid holding onto Doc/Proj/Solution, that would be good.

Reply:
Note that this _has_ been a problem before. This would manifest the same way as https://github.com/dotnet/roslyn/pull/50849.

Reply:
OK,  change it to workspace instead of Document